packages/PHPStanStaticTypeMapper/TypeMapper/UnionTypeMapper.php

```diff
@@ -518,6 +518,6 @@ private function narrowBoolType(
             return $phpParserUnionType;
         }
 
-        return $this->phpStanStaticTypeMapper->mapToPhpParserNode($unionType, $typeKind);
+        return null;
```

Comment:
this seems cause error test:

```diff

There was 1 failure:

1) Rector\Tests\Php80\Rector\FunctionLike\UnionTypesRector\UnionTypesRectorTest::test with data set #23
Failed on fixture file "narrow_bool_false2.php.inc"
Failed asserting that string matches format description.
--- Expected
+++ Actual
@@ @@
 
 class NarrowBoolFalse2
 {
-    public function go(bool|int $param): bool|int
+    /**
+     * @param bool|int|false $param
+     * @return bool|int|false
+     */
+    public function go($param)
```

I will check.

Reply:
skipping re-map on union type is safer, see comment https://github.com/rectorphp/rector-src/pull/3681#issuecomment-1520521976